app/uk/gov/hmrc/alcoholdutyreturns/service/AuditService.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.alcoholdutyreturns.service
+
+import com.google.inject.Inject
+import play.api.Logging
+import play.api.libs.json.{JsObject, Json, Writes}
+import uk.gov.hmrc.alcoholdutyreturns.config.AppConfig
+import uk.gov.hmrc.alcoholdutyreturns.models.audit.AuditEventDetail
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.audit.model.ExtendedDataEvent
+import uk.gov.hmrc.play.audit.AuditExtensions._
+import uk.gov.hmrc.play.audit.http.connector.AuditResult.Failure
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AuditService @Inject() (appConfig: AppConfig, auditConnector: AuditConnector)(implicit ec: ExecutionContext)
+    extends Logging {
+  def audit[T <: AuditEventDetail](detail: T)(implicit hc: HeaderCarrier, writes: Writes[T]): Future[Boolean] =
+    auditConnector
+      .sendExtendedEvent(
```

Comment:
Please call `def sendExplicitAudit[T](auditType: String, detail: T)` as it has a boilerplate code to deal with relevant audit fields like tags, auditSource, eventId

Reply:
This returns Unit which has two issues:
You can't log failure
You can't be certain that the audit event is complete before you exit the calls since the Future will just (hopefully!) run in background.

Reply:
Noted that audit is not crucial for legal cases so waiting is not going to be necessary anywhere. It would be worth checking to see if anything gets logged on failures, but hopefully (data stream being unresponsive will get caught elsewhere).
Changed to use the existing call.